apps/dolly-frontend/src/main/js/src/components/fagsystem/alderspensjon/form/validation.tsx

```diff
@@ -1,27 +1,60 @@
 import { ifPresent, messages, requiredNumber } from '@/utils/YupValidations'
 import * as Yup from 'yup'
-import { isAfter } from 'date-fns'
+import { isAfter, isBefore } from 'date-fns'
 
 export const validation = {
 	alderspensjon: ifPresent(
 		'$pensjonforvalter.alderspensjon',
 		Yup.object({
-			kravFremsattDato: Yup.date().nullable(),
+			kravFremsattDato: Yup.date()
+				.when('soknad', {
+					is: (soknad: Boolean) => !soknad,
+					then: () =>
+						Yup.date().test(
+							'is-before-iverksettelsesdato',
+							'Dato må være før iverksettelsesdato.',
+							function validate(kravFremsattDato) {
+								const iverksettelsesdato: Date =
+									this.options.context?.pensjonforvalter?.alderspensjon?.iverksettelsesdato
+								return kravFremsattDato && isBefore(kravFremsattDato, new Date(iverksettelsesdato))
+							},
+						),
+				})
+				.nullable(),
```

Comment:
Det ser ikke ut til at nullable() kicker inn her, for vi får en stygg feilmelding om vi tømmer feltet. Gjelder også for iverksettelsesdato. Ser ikke umiddelbart hva som er feil, men mulig de må flyttes litt på 🤔

Reply:
Vil fikse